packages/runtime-core/src/components/BaseTransition.ts

```diff
@@ -470,7 +471,8 @@ export function getTransitionRawChildren(
       )
     }
     // comment placeholders should be skipped, e.g. v-if
-    else if (keepComment || child.type !== Comment) {
+    // #4621, #4622, #4637 also skip text nodes
+    else if ((keepComment || child.type !== Comment) && child.type !== Text) {
```

Comment:
This is not a correct fix and will even cause the following code not to render `foo` correctly.
```html
<transition-group>
  foo
</transition-group>
```
We can't filter children here. The Transition depends on whether the content rendered by the child is an `Element` or not.